code/cfile/cfilearchive.cpp

```diff
@@ -116,12 +115,13 @@ int cftell( CFILE * cfile )
 	Assert(cb->fp != NULL);
 
 	#if defined(CHECK_POSITION) && !defined(NDEBUG)
-		int raw_position;
-		raw_position = ftell(cb->fp) - cb->lib_offset;
-		Assert(raw_position == cb->raw_position);
+	auto raw_position = ftell(cb->fp) - cb->lib_offset;
+	Assert(raw_position == cb->raw_position);
 	#endif
 
-	return cb->raw_position;
+	// The rest of the code still uses ints, do an overflow check to detect cases where this fails
+	Assertion(cb->raw_position <= std::numeric_limits<size_t>::max(), "Integer overflow in cftell, a file is probably too large (but I don't know which one).");
```

Comment:
I'm not sure I follow the logic of this Assertion. How does comparing with the maximum `size_t` guard against an `int` overflow?